packages/subgraph/schema.graphql

```diff
@@ -89,3 +89,88 @@ type CrabAuction @entity {
   owner: Bytes!
   timestamp: BigInt!
 }
+
+type CrabHedgeTimeThreshold @entity {
+  id: ID!
+  threshold: BigInt!
+  timestamp: BigInt!
+}
+
+type TimeLockTx @entity {
+  id: ID!
+  target: Bytes!
+  value: BigInt!
+  signature: String!
+  data: Bytes
+  eta: BigInt!
+  queued: Boolean
+  timestamp: BigInt!
+}
+
+type ExecuteTimeLockTx @entity {
+  id: ID!
+  timelocktx: TimeLockTx!
+  timestamp: BigInt!
+}
+
+type HedgeOTC @entity {
+  id: ID!
+  bidID: BigInt!
+  quantity: BigInt!
+  isBuying: Boolean!
+  clearingPrice: BigInt!
+  timestamp: BigInt!
+}
+
+type HedgeOTCSingle @entity {
+  id: ID!
+  trader: Bytes!
+  bidID: BigInt!
+  quantity: BigInt!
+  price: BigInt!
+  isBuying: Boolean!
+  clearingPrice: BigInt!
+  hedgeOTC: HedgeOTC!
+  timestamp: BigInt!
+}
+
+type SetStrategyCap @entity {
+  id: ID!
+  cap: BigInt!
+  timestamp: BigInt!
+}
+
+type SetHedgingTwapPeriod @entity {
+  id: ID!
+  hedging: BigInt!
+  timestamp: BigInt!
+}
+
+type SetHedgePriceThreshold @entity {
+  id: ID!
+  threshold: BigInt!
+  timestamp: BigInt!
+}
+
+type SetOTCPriceTolerance @entity {
+  id: ID!
+  tolerance: BigInt!
+  timestamp: BigInt!
+}
+
+type VaultTransferred @entity {
+  id: ID!
+  vaultID: BigInt!
+  strategy: Bytes!
+  timestamp: BigInt!
+}
+
+type CrabUserTx @entity {
+  id: ID!
+  owner: Bytes!
+  user: Bytes!
```

Comment:
@KMKoushik Should we create a user entity , and make all these tx's children of the user entity?
Also should we maintain a running account balances in that entity? Does it help for frontend or other analytics ; PnL?

Reply:
It would be a nice but not important IMO. We use on-chain balance to infer PnL to be sure!